mh/time_utils.py

```diff
@@ -21,8 +21,72 @@
 
 import datetime
 
+
 EPOCH = datetime.datetime(1970, 1, 1)
+TICKS_PER_CYCLE = 23040  # 6.4 hours per cycle, 3.2 hours per daytime/nighttime
 
 
 def datetime_to_int(dt):
     return int((dt - EPOCH).total_seconds())
+
+SECONDS_PER_DAY = 24*60*60
+JHEN_EVENT_OFFSET = 14  # Cycle of 14 (real) days
+FOG_START = 0  
+JHEN_START = 1  # 1 day of fog
+JHEN_END = 3  # Followed by 2 (real) days of sandstorm
+
+
+def current_tick():
+    """
+    This product of this function is based on the assumption that
+    any decreasing integer value is sufficient for tracking the
+    passage of time in Monster Hunter Tri.
+    
+    Debugging has shown that the internal time value (initialized
+    via a request to the server which responds using this function)
+    continuously decreases by one per second, and an internal modulus
+    is used to turn the ever-decreasing value into a point in the
+    day/night cycle (with a cycle length specified by the server).
+    
+    It is also suspected that, should this value ever reach zero, an
+    integer underflow happens, which disrupts the day/night cycle.
+    
+    As a result, to ensure that the underflow never happens, the current
+    implementation uses a reversed epoch time that counts down from
+    UINT_MAX (the game's internal maximum for this value) as the seconds
+    pass. This also has the effect of time "passing" even when the
+    server is shut down.
+    """
+    return 0xFFFFFFFF - datetime_to_int(datetime.datetime.now())
+
+
+def current_server_time():
+    return datetime_to_int(datetime.datetime.now())
+
+
+def get_jhen_cycle_time(curr_time=None):
```

Comment:
This function is only used here and doesn't make much sense on its own. You might want to move its content into `get_jhen_event_times` directly.

Reply:
I think there's value in the readability that comes with separating this logic out into another function, and if we want to base other event quest timings in the cycle on this one "starting" time we're going to want it separated out. Right now it's just the Jhen quest that we're timing based on this cycle, but in the future we're going to need some sort of rotation, and I think having this separated out will help with that when we get there.

Reply:
Actually I take that back, this is something that should be worried about once we actually implement the rest of the event quest cycle system.